src/lib/pubkey/kyber/kyber_common/kyber_keys.h

```diff
@@ -60,37 +58,49 @@ class Kyber_PublicKeyInternal {
 
    private:
       const KyberConstants m_mode;
-      KyberPolyVecNTT m_t;
-      const KyberSeedRho m_rho;
       const KyberSerializedPublicKey m_public_key_bits_raw;
       const KyberHashedPublicKey m_H_public_key_bits_raw;
+      KyberPolyVecNTT m_t;
+      const KyberSeedRho m_rho;
 };
 
 class Kyber_PrivateKeyInternal {
    public:
-      Kyber_PrivateKeyInternal(KyberConstants mode, KyberPolyVecNTT s, KyberImplicitRejectionValue z) :
-            m_mode(std::move(mode)), m_s(std::move(s)), m_z(std::move(z)) {}
+      Kyber_PrivateKeyInternal(KyberConstants mode, KyberPolyVecNTT s, KyberPrivateKeySeed seed) :
+            m_mode(std::move(mode)), m_s(std::move(s)), m_seed(std::move(seed)) {}
 
       KyberMessage indcpa_decrypt(StrongSpan<const KyberCompressedCiphertext> ct) const;
 
       KyberPolyVecNTT& s() { return m_s; }
 
       const KyberPolyVecNTT& s() const { return m_s; }
 
-      const KyberImplicitRejectionValue& z() const { return m_z; }
+      const KyberPrivateKeySeed& seed() const { return m_seed; }
+
+      const KyberImplicitRejectionValue& z() const { return m_seed.z; }
 
       const KyberConstants& mode() const { return m_mode; }
 
       Kyber_PrivateKeyInternal() = delete;
 
-      void _const_time_poison() const { CT::poison_all(m_s, m_z); }
+      void _const_time_poison() const {
+         CT::poison_all(m_s, m_seed.z);
+         if(m_seed.d.has_value()) {
```

Comment:
We might want to consider a variant of `poison` that takes an `optional<T>` and poisons it if it has a value, as a convenience. (No need to do so in this PR)

Reply:
Sounds like a good idea. I'll look at this independently.